pinot-common/src/main/java/org/apache/pinot/common/response/PinotBrokerTimeSeriesResponse.java

```diff
@@ -0,0 +1,168 @@
+    return _errorType;
+  }
+
+  public String getError() {
+    return _error;
+  }
+
+  public String serialize()
+      throws JsonProcessingException {
+    return OBJECT_MAPPER.writeValueAsString(this);
+  }
+
+  public static PinotBrokerTimeSeriesResponse newSuccessResponse(Data data) {
+    return new PinotBrokerTimeSeriesResponse(SUCCESS_STATUS, data, null, null);
+  }
+
+  public static PinotBrokerTimeSeriesResponse newErrorResponse(String errorType, String errorMessage) {
+    return new PinotBrokerTimeSeriesResponse(ERROR_STATUS, Data.EMPTY, errorType, errorMessage);
+  }
+
+  public static PinotBrokerTimeSeriesResponse fromTimeSeriesBlock(TimeSeriesBlock seriesBlock) {
+    if (seriesBlock.getTimeBuckets() == null) {
+      throw new UnsupportedOperationException("Non-bucketed series block not supported yet");
+    }
+    return convertBucketedSeriesBlock(seriesBlock);
+  }
+
+  private static PinotBrokerTimeSeriesResponse convertBucketedSeriesBlock(TimeSeriesBlock seriesBlock) {
+    Long[] timeValues = Objects.requireNonNull(seriesBlock.getTimeBuckets()).getTimeBuckets();
+    List<PinotBrokerTimeSeriesResponse.Value> result = new ArrayList<>();
+    for (var listOfTimeSeries : seriesBlock.getSeriesMap().values()) {
+      Preconditions.checkState(!listOfTimeSeries.isEmpty(), "Received empty time-series");
+      TimeSeries anySeries = listOfTimeSeries.get(0);
+      // TODO: Ideally we should allow "aliasing" a series, and hence we should store something like a series-name.
+      //  Though in most contexts that would serve the same purpose as an ID.
+      Map<String, String> metricMap = new HashMap<>();
+      metricMap.put(METRIC_NAME_KEY, anySeries.getTagsSerialized());
+      metricMap.putAll(anySeries.getTagKeyValuesAsMap());
+      for (TimeSeries timeSeries : listOfTimeSeries) {
+        Object[][] values = new Object[timeValues.length][];
+        for (int i = 0; i < timeValues.length; i++) {
+          Object nullableValue = timeSeries.getValues()[i];
+          values[i] = new Object[]{timeValues[i], nullableValue == null ? null : nullableValue.toString()};
+        }
+        result.add(new PinotBrokerTimeSeriesResponse.Value(metricMap, values));
+      }
+    }
+    PinotBrokerTimeSeriesResponse.Data data = PinotBrokerTimeSeriesResponse.Data.newMatrix(result);
+    return PinotBrokerTimeSeriesResponse.newSuccessResponse(data);
+  }
+
+  public static class Data {
+    public static final Data EMPTY = new Data("", new ArrayList<>());
+    private final String _resultType;
+    private final List<Value> _result;
+
+    @JsonCreator
+    public Data(@JsonProperty("resultType") String resultType, @JsonProperty("result") List<Value> result) {
+      _resultType = resultType;
+      _result = result;
+    }
+
+    public String getResultType() {
+      return _resultType;
+    }
+
+    public List<Value> getResult() {
+      return _result;
+    }
+
+    public static Data newMatrix(List<Value> result) {
```

Comment:
We will need to add other types "vector", "scalar" , "string" as well for prometheus.